docs/_sources/routes/direct-storeandnotify.rst.txt

```diff
@@ -0,0 +1,54 @@
+Direct Store and Notify
+***********************
+
+Purpose
+========
+The Linux for Health (LFH) direct:storeandnotify route allows you to store transaction data as part of the LFH Longitudinal Patient Record (LPR) and notify listeners of the stored data via NATS for downstream integration.
+
+Details
+=======
++-------------------------+---------------------------------------------------------------------+
+| Attribute               | Value                                                               |
++=========================+=====================================================================+
+| Route Name              | direct:storeandnotify                                               |
++-------------------------+---------------------------------------------------------------------+
+| Supported Operation(s)  | Camel direct invocation                                             |
++-------------------------+---------------------------------------------------------------------+
+
+Configuration
+=============
+The direct:storeandnotify route expects the following input state in the Camel message Exchange.
+
+Exchange State
+--------------
++-------------------------+--------------------------------------------------------------------+
+| Element                 | Description                                                        |
++=========================+====================================================================+
+| Message Properties      | |msgconfig_def|                                                    |
++-------------------------+---------------+----------------------------------------------------+
+| Message Body            | |msgbody_def|                                                      |
++-------------------------+---------------+----------------------------------------------------+
+
+.. |msgconfig_def| replace:: LFH message properties are stored as Exchange properties.  Before direct:storeandnotify is called, these Exchange properties must be set via the LinuxForHealthMessageProcessor or a similar processor. This ensures that the message will be stored in the data store using the LFH standard format.
```

Comment:
Are these LFH exchange properties enumerated somewhere in the doc?
One could figure this out if they look at the `LinuxForHealthMessage.java` class/file, but it'd be nice to have this in the doc with a description/example of each property. I wasn't quite sure what each field should be set to when I did this for ACD.
**Note:** the `LinuxForHealthMessage.java` file is the only file in the *.processor java package that does not have a **Processor** suffix. We may want to update the class/filename with this suffix or update the "LinuxForHealthMessageProcessor" reference above to omit the **Processor* suffix.

Reply:
At a high level, they are in message-structure.html here: https://linuxforhealth.github.io/docs/message-structure.html.  But more to your point, Dixon is fixing this by creating one processor that doesn't need to be modified that will set the message properties.  This was possible after the recent LFH property changes.  That new processor will probably be sufficient to avoid confusion for others, I think.